pkg/util/limiter/query_limiter.go

```diff
@@ -94,9 +95,9 @@ func (ql *QueryLimiter) AddSeries(seriesLabels labels.Labels) validation.LimitEr
 			ql.queryMetrics.QueriesRejectedTotal.WithLabelValues(stats.RejectReasonMaxSeries).Inc()
 		}
 
-		return NewMaxSeriesHitLimitError(uint64(ql.maxSeriesPerQuery))
+		return duplicated, NewMaxSeriesHitLimitError(uint64(ql.maxSeriesPerQuery))
```

Comment:
```suggestion
		return seriesLabels, NewMaxSeriesHitLimitError(uint64(ql.maxSeriesPerQuery))
```

Reply:
The suggestion should already be applied in the latest code.